src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,180 @@
+package duke;
+
+import java.time.format.DateTimeParseException;
+
+import duke.command.AddCommand;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+
+import duke.exception.DukeEmptyInputException;
+import duke.exception.DukeInvalidCommandException;
+import duke.exception.DukeInvalidDateTimeException;
+import duke.exception.DukeInvalidKeywordException;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+/**
+ * Responsible for parsing user inputs.
+ */
+public class Parser {
+
+    /**
+     * Parses the user input and returns a Command based on the input.
+     *
+     * @param fullCommand User input.
+     * @throws DukeEmptyInputException If input is empty.
+     * @throws DukeInvalidDateTimeException If date and time input is invalid.
+     * @throws DukeInvalidCommandException If command is invalid.
+     */
+    static Command parse(String fullCommand) throws DukeEmptyInputException, DukeInvalidDateTimeException,
+            DukeInvalidCommandException, DukeInvalidKeywordException {
+        String[] commandArr = fullCommand.trim().split(" ", 2);
```

Comment:
Consider changing "commandArr" to "commands"? Coding standards say collections should be in plural form.